packages/studio-plugin/src/startPagesDevelopmentServer.ts

```diff
@@ -0,0 +1,53 @@
+import { spawn, execSync } from "node:child_process";
+
+export const startPagesDevelopmentServer = async () => {
+  const pagesServer = spawn("npx", ["pages", "dev", "--open-browser=false"], {
+    stdio: ["pipe", "pipe", "pipe"],
+    env: {
+      ...process.env,
+    },
+  });
+
+  // stream pages server errors to parent process with a prefix
+  pagesServer.stderr.setEncoding("utf-8");
+  pagesServer.stderr.on("data", (data: string) => {
+    data.split("\n").forEach((line: string) => {
+      console.error("Pages Development Server |", line.trim());
+    });
+  });
+
+  // kill studio if the pages server exits
+  pagesServer.addListener("exit", () => {
+    process.exit(-1);
+  });
+
+  // kill pages server child process when studio exits
+  // by getting the dev server port on startup and then
+  // killing the proceess running at that port
+  const port : number | void = await new Promise(
+    (resolve) => {
+      pagesServer.stdout.setEncoding("utf-8");
+      pagesServer.stdout.on("data", (data: string) => {
+        const matches = data.match(/listening on :(\d+)/);
+        if (matches && matches.length >= 2) {
+          const port = Number.parseInt(matches[1])
+          if (isNaN(port)) {
+            resolve()
+          }
+          resolve(port);
+        }
+        setTimeout(resolve, 10000)
+      });
+    }
+  );
+
+  // kill pages server when studio exits
+  process.on("exit", () => {
+    if (port) {
+      console.log("Studio is shutting down")
+      execSync(`npx kill-port ${port}`)
```

Comment:
you don't need to exec this, you can directly call import the package and call it

Reply:
The issue here (and why I'm still doing `await startPagesDevelopmentServer();` is that `process.on("exit"`... doesn't reliably work with async code. The kill-port package is an async function and reading the port from the server's stdout also has to be async